src/object/interrupt.c

```diff
@@ -295,3 +295,18 @@ void setIRQState(irq_state_t irqState, irq_t irq)
 #endif
     maskInterrupt(irqState == IRQInactive, irq);
 }
+
+void maybeHandleInterrupt(void)
+{
+    irq_t irq;
+
+    irq = getActiveIRQ();
+    if (IRQT_TO_IRQ(irq) != IRQT_TO_IRQ(irqInvalid)) {
+        handleInterrupt(irq);
+    } else {
+#ifdef CONFIG_IRQ_REPORTING
+        userError("Spurious interrupt!");
+#endif
+        handleSpuriousIRQ();
```

Comment:
The resetting IRQs that am335x and omap3 do does seem like a dubious thing to do, except if it works around an errata, but no errata is mentioned.

Reply:
I've been wondering about that as well. I'll try to find the commit that introduces them, maybe there is some information there.

Reply:
It does seem to at least not be an accident: [SELFOUR-154](https://sel4.atlassian.net/browse/SELFOUR-154)

Reply:
Ah, in OMAP's case it is just acking the IRQ. Probably the same for am335x too. So just misleading comments then.